### PR TITLE
Enable DGRAD RS overlap

### PR DESCRIPTION
**Pull request by minitu**

### CODE REVIEW

**Comment by minitu** [comment]
Corresponding NeMo/MCore PRs:
- NeMo: https://github.com/NVIDIA/NeMo/pull/8840
- MCore: https://gitlab-master.nvidia.com/ADLR/megatron-lm/-/merge_requests/1320

File: transformer_engine/pytorch/module/base.py [hunk 1]
```diff
@@ -130,12 +130,14 @@ def initialize_ub(
...
     ]
     layers_reduce_scatter_overlap = ["proj_fprop", "fc2_fprop", "qkv_wgrad", "fc1_wgrad"]
+    dgrad_reduce_scatter_overlap = ["qkv_dgrad", "fc1_dgrad"]
     # Default overlap methods for layers
     methods = {
         "ring_exchange":["qkv_fprop", "fc1_fprop", "proj_dgrad", "fc2_dgrad"],
         "pipeline":["proj_fprop", "fc2_fprop"],
         "bulk":["qkv_dgrad", "qkv_wgrad", "fc1_dgrad", "fc1_wgrad"],
     }
+    layers_reduce_scatter_overlap = ["proj_fprop", "fc2_fprop", "qkv_wgrad", "fc1_wgrad"]
```
Comment (timmoon10) [inline transformer_engine/pytorch/module/base.py:140]: Nit: this line is repeated.
```suggestion
```
https://github.com/NVIDIA/TransformerEngine/blob/f329a7253756be9326a93649c49fefbd2d27a45e/transformer_engine/pytorch/module/base.py#L132

File: transformer_engine/pytorch/module/layernorm_linear.py [hunk 2]
```diff
@@ -428,14 +462,14 @@ def backward(
...
-                if ctx.ub_bulk_wgrad and ub_obj_dgrad.is_fp8_ubuf():
+                if (ctx.ub_bulk_wgrad or ctx.ub_overlap_rs_dgrad) and ub_obj_dgrad.is_fp8_ubuf():
                     out_index = tex.FP8BwdTensors.GRAD_INPUT1
                     meta_tensor = ctx.fp8_meta["scaling_bwd"]
                     out_te_type = fp8_dtype_backward
                     out_type = torch.uint8
                     ub_obj_dgrad.set_ubuf_scale_inv(meta_tensor.scale_inv[out_index])
 
-                # DGRAD: Evaluated unconditionally to feed into Linear backward
+                # DGRAD: Evaluated unconditionally to feed into Linear backward vasu
```
Comment (timmoon10) [inline transformer_engine/pytorch/module/layernorm_linear.py:472]: Nit:
```suggestion
                # DGRAD: Evaluated unconditionally to feed into Linear backward
```

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**